### PR TITLE
context time for graphql changes dynamically

### PR DESCRIPTION
**Pull request by prithvi081099**

### CODE REVIEW

File: gateway/utils/handlers/graphql.go [hunk 1]
```diff
@@ -9,21 +9,40 @@ import (
...
-		// Create a context of execution
-		ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
-		defer cancel()
+		c := new(config.Project)
+		err := json.NewDecoder(r.Body).Decode(c)
+		defer utils.CloseTheCloser(r.Body)
+
+		// Throw error if request was of incorrect type
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]: use logrus.Errorf()
to log the errors
for reference
https://github.com/spaceuptech/space-cloud/blob/v0.16.0/runner/server/handle.go

File: gateway/utils/handlers/graphql.go [hunk 2]
```diff
@@ -9,21 +9,40 @@ import (
...
 	"github.com/gorilla/mux"
 
+	"github.com/spaceuptech/space-cloud/gateway/config"
 	"github.com/spaceuptech/space-cloud/gateway/model"
 	"github.com/spaceuptech/space-cloud/gateway/utils"
 	"github.com/spaceuptech/space-cloud/gateway/utils/graphql"
+	"github.com/spaceuptech/space-cloud/gateway/utils/syncman"
 )
 
 // HandleGraphQLRequest creates the graphql operation endpoint
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]:
```suggestion
// HandleGraphQLRequest executes graphql queries
```

File: gateway/utils/handlers/graphql.go [hunk 3]
```diff
@@ -9,21 +9,40 @@ import (
...
+			w.WriteHeader(http.StatusBadRequest)
+			_ = json.NewEncoder(w).Encode(map[string]string{"error": "Config was of invalid type - " + err.Error()})
+			return
+		}
 
 		vars := mux.Vars(r)
 		projectID := vars["project"]
+
+		projectConfig, err := syncMan.GetConfig(projectID)
+		if err != nil {
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]: set status code & return error to client also log the error using logrus

File: gateway/utils/handlers/graphql.go [hunk 4]
```diff
@@ -61,7 +80,7 @@ func HandleGraphQLRequest(graphql *graphql.Module) http.HandlerFunc {
 		select {
 		case <-ch:
 			return
-		case <-time.After(10 * time.Second):
+		case <-time.After(time.Duration(projectConfig.ContextTime) * time.Second):
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]: @YourTechBud is this correct ?

File: gateway/config/config.go [hunk 5]
```diff
@@ -9,11 +9,12 @@ type Config struct {
...
-	AESkey  string   `json:"aesKey" yaml:"aesKey"`
-	ID      string   `json:"id" yaml:"id"`
-	Name    string   `json:"name" yaml:"name"`
-	Modules *Modules `json:"modules" yaml:"modules"`
+	Secret      string   `json:"secret" yaml:"secret"`
+	AESkey      string   `json:"aesKey" yaml:"aesKey"`
+	ID          string   `json:"id" yaml:"id"`
+	Name        string   `json:"name" yaml:"name"`
+	Modules     *Modules `json:"modules" yaml:"modules"`
+	ContextTime int      `json:"contextTime" yaml:"contextTime"`
```
Comment (sharadregoti) [inline gateway/config/config.go]: @YourTechBud should we add comment to these individual fields ?
specifying what it does

Comment (YourTechBud) [reply gateway/config/config.go]: Just on the context time field since it isn't super clear

File: gateway/utils/handlers/graphql.go (hunk 4 above)
Comment (YourTechBud) [reply gateway/utils/handlers/graphql.go]: Yes

File: gateway/utils/handlers/graphql.go [hunk 6]
```diff
@@ -8,22 +8,44 @@ import (
...
-		// Create a context of execution
-		ctx, cancel := context.WithTimeout(r.Context(), 10*time.Second)
-		defer cancel()
+		c := new(config.Project)
+		err := json.NewDecoder(r.Body).Decode(c)
+		defer utils.CloseTheCloser(r.Body)
+
+		// Throw error if request was of incorrect type
+		if err != nil {
+			logrus.Errorf("Config was of invalid type - - %s", err.Error())
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]:
```suggestion
			logrus.Errorf("Error handling graphql query execution unable to decode request body - %s", err.Error())
```

File: gateway/utils/handlers/graphql.go [hunk 7]
```diff
@@ -8,22 +8,44 @@ import (
...
+			utils.SendErrorResponse(w, r, http.StatusBadRequest, err)
+			return
+		}
 
 		vars := mux.Vars(r)
 		projectID := vars["project"]
+
+		projectConfig, err := syncMan.GetConfig(projectID)
+		if err != nil {
+			logrus.Errorf("Config was of invalid type - - %s", err.Error())
```
Comment (sharadregoti) [inline gateway/utils/handlers/graphql.go]:
```suggestion
			logrus.Errorf("Error handling graphql query execution unable to get project config of %s - %s",projectID, err.Error())
```

**Pull request closed without merging**